cpp/csp/python/PyPushPullInputAdapter.cpp

```diff
@@ -119,12 +119,26 @@ struct PyPushPullInputAdapter_PyObject
         CSP_RETURN_NONE;
     }
 
+    static PyObject * engineShutdown( PyPushPullInputAdapter_PyObject * self, PyObject * args, PyObject * kwargs )
+    {
+        CSP_BEGIN_METHOD;
+
+        const char* msg;
+        PyArg_ParseTuple( args, "s", &msg );
```

Comment:
We have 3 copies of this function (`PyAdapterManager`, `PyPushInput` and `PyPushPullInput`). Let's just make one in `PyEngine.h` (with a template arg for self) and re-use it for each of the 3 classes.